### PR TITLE
Support validation error Message detail

### PR DESCRIPTION
**Pull request by shuheiktgw**
I've added support for the Message type error details.

### CODE REVIEW

File: resolver/resolver.go [hunk 1]
```diff
@@ -1265,6 +1267,35 @@ func (r *Resolver) resolveMessageRuleValidationDetails(details []*federation.Val
 	return result
 }
 
+func (r *Resolver) resolveValidationDetailMessages(ctx *context, messages []*federation.MessageExpr) VariableDefinitions {
+	if len(messages) == 0 {
+		return nil
+	}
+	msgs := make([]*federation.VariableDefinition, 0, len(messages))
+	for idx, message := range messages {
+		name := fmt.Sprintf("_def%d_err_detail%d_msg%d", ctx.defIndex(), ctx.errDetailIndex(), idx)
```
Comment (shuheiktgw) [inline resolver/resolver.go:1276]: This feels a bit too hacky but I cannot think of a better way...

File: _examples/12_simple_validation/proto/federation/federation.proto [hunk 2]
```diff
@@ -36,24 +44,44 @@ message GetPostResponse {
...
+                    description: "'description2'"
+                  }
+                ]
               }
               localized_message {
                 locale: "en-US"
                 message: "post.content"
               }
-            }
+            },
```
Comment (goccy) [inline _examples/12_simple_validation/proto/federation/federation.proto:84]: Is this comma unnecessary ?

File: resolver/context.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ type context struct {
 	defIdx       int
 	depIdx       int
 	argIdx       int
+	errDetailIdx int
```
Comment (goccy) [inline resolver/context.go:15]: Please add `errDetailIdx` field to [`clone()`](https://github.com/mercari/grpc-federation/pull/68/files#diff-2404bd4fa6ff38bb60e2e9c2e51a79c8f5a5d6a9135ae29ca6bf393d6d875020R31) method

File: testdata/validation.proto [hunk 4]
```diff
@@ -36,6 +36,22 @@ message GetPostResponse {
             code: FAILED_PRECONDITION
             details {
               rule: "post.title == 'some-title'"
+              message: [
+                {
+                  name: "CustomMessage",
+                  args: {
+                    name: "message",
+                    by: "'message1'"
```
Comment (goccy) [inline testdata/validation.proto:44]: I think it is simpler to write `string: "message1"`

File: validator/testdata/invalid_validation_message_argument.proto [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        name: "res"
+        call {
+          method: "post.PostService/GetPost"
+          request { field: "id", by: "$.id" }
+        }
+      },
+      { name: "post", by: "res.post", autobind: true },
+      {
+        validation {
+          name: "invalidMessageArgument",
```
Comment (goccy) [inline validator/testdata/invalid_validation_message_argument.proto:47]: Is there an intention behind the name lowerCamelCase ?
In the example, everything else is lower_snake_case.

Comment (shuheiktgw) [reply validator/testdata/invalid_validation_message_argument.proto:47]: Not much! Let me fix that

**Review by goccy — APPROVED** [review]
LGTM !!

**Pull request merged**